pkg/hubble/metrics/http/handler.go

```diff
@@ -69,50 +74,106 @@ func (h *httpHandler) Status() string {
 	if h.context == nil {
 		return ""
 	}
-	return h.context.Status()
+	return h.context.Status() + fmt.Sprintf(",exemplars=%t", h.exemplars)
 }
 
 func (h *httpHandler) ProcessFlow(ctx context.Context, flow *flowpb.Flow) error {
-	l7 := flow.GetL7()
+	if h.useV2 {
+		return h.processMetricsV2(flow)
+	} else {
+		return h.processMetricsV1(flow)
+	}
+}
+
+func (h *httpHandler) getMetadataFromFlow(flow *flowpb.Flow) (l7 *flowpb.Layer7, http *flowpb.HTTP, reporter string, traceID string, ok bool) {
+	l7 = flow.GetL7()
 	if l7 == nil {
-		return nil
+		return
 	}
-	http := l7.GetHttp()
+	http = l7.GetHttp()
 	if http == nil {
-		return nil
+		return
+	}
+	if l7.GetType() != flowpb.L7FlowType_REQUEST &&
+		l7.GetType() != flowpb.L7FlowType_RESPONSE {
+		return
```

Comment:
Proto generated Go code is nil safe. Therefore, this code section can be simplified into something like this:
```go
        switch flow.GetL7().GetHttp().GetType() {
        case flowpb.L7FlowType_REQUEST,
                flowpb.L7FlowType_RESPONSE:
        default:
                return
        }
```

Reply:
Ah, is just moving exist code, but previously the intermediate variables were used in more places to be less verbose, so I suppose we could reduce this down now.

Reply:
Actually, to note: the suggestion and the original code aren't equivalent. There's no `type` on the HTTP field, it's on L7. We're checking the type && we're checking for non-empty HTTP. Either way I'll be making adjustments here.

Reply:
Refactored. PTAL.